src/test/scala/uk/gov/nationalarchives/tdr/api/service/ClientFileMetadataServiceSpec.scala

```diff
@@ -26,40 +27,38 @@ class ClientFileMetadataServiceSpec extends AnyFlatSpec with MockitoSugar with M
     val dummyInstant = Instant.now()
     val dummyTimestamp = Timestamp.from(dummyInstant)
     val dummyFileSize: Long = 1000
-    val repositoryMock = mock[ClientFileMetadataRepository]
-    val mockResponse = Future.successful(Seq(ClientfilemetadataRow(
-      clientMetadataUuid,
-      fileUuid,
-      Some("dummy/original/path"),
-      Some("dummyCheckSum"),
-      Some("checksumType"),
-      dummyTimestamp,
-      Some(dummyFileSize),
-      dummyTimestamp
+    val repositoryMock = mock[FileMetadataRepository]
+    def row(name: String, value: String): FilemetadataRow =
+      FilemetadataRow(clientMetadataUuid, fileUuid, Option.empty, value, Timestamp.from(FixedTimeSource.now), UUID.randomUUID(), Option(name))
+
+    val mockResponse = Future(Seq(
+      row(ClientSideFileSize, dummyFileSize.toString),
+      row(ClientSideFileLastModifiedDate, dummyTimestamp.toString),
+      row(ClientSideOriginalFilepath, "dummy/original/path"),
+      row(SHA256ClientSideChecksum, "dummyCheckSum")
     ))
-    )
 
-    when(repositoryMock.addClientFileMetadata(any[Seq[ClientfilemetadataRow]])).thenReturn(mockResponse)
-    val service = new ClientFileMetadataService(repositoryMock, fixedUuidSource)
+    when(repositoryMock.getFileMetadata(fileUuid, clientSideProperties:_*)).thenReturn(mockResponse)
+    when(repositoryMock.addFileMetadata(any[Seq[FilemetadataRow]])).thenReturn(mockResponse)
+
+    val service = new ClientFileMetadataService(repositoryMock, fixedUuidSource, FixedTimeSource)
     val result = service.addClientFileMetadata(Seq(AddClientFileMetadataInput(
       fileUuid,
       Some("dummy/original/path"),
       Some("dummyCheckSum"),
-      Some("checksumType"),
+      Some("SHA256"),
       dummyInstant.toEpochMilli,
       Some(dummyFileSize),
-      dummyInstant.toEpochMilli))).futureValue
+      Option(dummyInstant.toEpochMilli))), UUID.randomUUID()).futureValue
 
     result.length shouldBe 1
     val r: ClientFileMetadata = result.head
     r.fileId shouldBe fileUuid
     r.originalPath.get shouldBe "dummy/original/path"
     r.checksum.get shouldBe "dummyCheckSum"
-    r.checksumType.get shouldBe "checksumType"
+    r.checksumType.get shouldBe "SHA256"
```

Comment:
Maybe check data added to the DB as part of the test?

Reply:
Because this is the service spec, the repository calls are mocked so nothing gets added to the database by the test. I think this is fine.